### PR TITLE
Numbers as subject

### LINKED ISSUE #114: Error when the nominal subject is a number (e.g. year 1872, "two", etc.)
whoami returns "Subject not found" errors when the nominal subject of the copular verb is not a noun. Technically, numbers can be nouns, though spaCy marks them as numbers, which is why we get the error in whoami. For example

a. 1066 is a famous year.
b. 1066, the year that the Normans conquered England, is a famous year.

a returns an error. There is no error in b because spaCy thinks "year" (as in, "the year that the Normans conquered England") is the nominal subject rather than 1066. This is probably a separate issue that has to do with a parsing bug in spaCy. 

In any case, how should we deal with numbers that are nouns? Initially I am thinking that we can allow whoami to accept both nouns and numbers as nominal subjects. However, I can imagine how this might cause some issues. Let's try looking into this method and other ways to handle the issue.

### PR DESCRIPTION
**Pull request by shu98**
Resolves #114. The approach is to treat numbers/dates (which are nominal subjects of the root be-verb) as valid sentence subjects.

### CODE REVIEW

File: whoami/defextract.py [hunk 1]
```diff
@@ -132,11 +132,11 @@ def get_subject(head):
         if sdep.is_clausal_subject(child):
             return get_subject(child)
         elif sdep.is_direct_subject(child):
-            if not spos.is_noun(child):
+            if not spos.is_noun(child) and not spos.is_number(child) and not spos.is_date(child) and child not in ners:
```
Comment (pondpremt) [inline whoami/defextract.py]: lets do: 

```
if spos.is_noun(child) or spos.is_number(child) or ... :
    return child
return ParseError(..)
```

File: whoami/defextract.py [hunk 2]
```diff
@@ -132,7 +132,7 @@ def get_subject(head):
         if sdep.is_clausal_subject(child):
             return get_subject(child)
         elif sdep.is_direct_subject(child):
-            if spos.is_noun(child) or spos.is_number(child) or spos.is_date(child):
```
Comment (hayleybergman) [inline whoami/defextract.py]: What was the reason for removing `spos.is_date(child)`?

Comment (hayleybergman) [inline whoami/defextract.py]: What was the reason for removing `spos.is_date(child)`?

Comment (hayleybergman) [inline whoami/defextract.py]: What was the reason for removing `spos.is_date(child)`?

Comment (hayleybergman) [reply whoami/defextract.py]: Makes sense, looks good to me then!

**Pull request merged**